easybuild/framework/easyconfig/tweak.py

```diff
@@ -88,17 +90,21 @@ def tweak(easyconfigs, build_specs, modtool, targetdirs=None):
                              toolchains)
     # Toolchain is unique, let's store it
     source_toolchain = easyconfigs[-1]['ec']['toolchain']
-    modifying_toolchains = False
+    modifying_toolchains_or_deps = False
     target_toolchain = {}
     src_to_dst_tc_mapping = {}
     revert_to_regex = False
+    if 'upgrade_deps' in build_specs:
+        upgrade_dependencies = build_specs['upgrade_deps']
+    else:
+        upgrade_dependencies = False
 
-    if 'toolchain_name' in build_specs or 'toolchain_version' in build_specs:
+    if 'toolchain_name' in build_specs or 'toolchain_version' in build_specs or upgrade_dependencies:
         keys = build_specs.keys()
 
         # Make sure there are no more build_specs, as combining --try-toolchain* with other options is currently not
         # supported
-        if any(key not in ['toolchain_name', 'toolchain_version', 'toolchain'] for key in keys):
+        if any(key not in ['toolchain_name', 'toolchain_version', 'toolchain', 'upgrade_deps'] for key in keys):
             print_warning("Combining --try-toolchain* with other build options is not fully supported: using regex")
```

Comment:
@ocaisa Warning message should be update to mention `--try-toolchain* or --try-upgrade-deps`?